numpy/ma/core.py

```diff
@@ -3105,6 +3105,14 @@ def __getitem__(self, indx):
         Return the item described by i, as a masked array.
 
         """
+        # 2016.01.15 -- v1.11.0
+        warnings.warn(
+            "Currently, slicing will try to return a view of the data," +
+            " but will return a copy of the mask. In the future, it will try" +
+            " to return both as views.",
+            FutureWarning
+        )
+
```

Comment:
> Do we need warnings in `__getitem__`?

The return value is changing. Personally, I think that merits a warning.

> The proposed change shouldn't have any effect on someone who never modifies the mask, right?

Could the change not effect them? Maybe, but I'd rather they start thinking about it now instead of leaving them in the dark until the release, and then discovering some behavior is broken.

> Also, perhaps we can be even fancier and only show the warning if we know `self.data` is a view...

I suppose, but I think we have just as much chance of getting this wrong by being too tricky. Namely, not emitting a warning when we should. The current behavior is straightforward. If the user calls `__setitem__`, they are always warned about the breaking change that is pending.